### PR TITLE
Add support for cans12erv2

### PR DESCRIPTION
**Pull request by canbaytok**
## Description
I've worked on a new version of my small handwired DIY keypad, which now features hot swap switch sockets and an encoder knob.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/cans12erv2/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+
+#define _KEYPAD 0
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [_KEYPAD] = LAYOUT(
```
Comment (waffle87) [inline keyboards/handwired/cans12erv2/keymaps/default/keymap.c:13]:
```suggestion
const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
    [0] = LAYOUT(
```

File: keyboards/handwired/cans12erv2/keymaps/default/rules.mk [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ENCODER_ENABLE = yes
```
Comment (waffle87) [inline keyboards/handwired/cans12erv2/keymaps/default/rules.mk:1]:
```suggestion
```

File: keyboards/handwired/cans12erv2/readme.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Cans12erV2
+![Cans12erV2](https://raw.githubusercontent.com/canbaytok/Cans12erV2/refs/heads/main/assets/Cans12erV2.png)
```
Comment (waffle87) [inline keyboards/handwired/cans12erv2/readme.md:2]: This image should be rehosted on Imgur.

File: keyboards/handwired/cans12erv2/readme.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+New version of my 12-key orthogonal macro keypad with hot swappable keyswitch support and an encoder knob.
+Programming is planned to be possible with [QMK/VIA](https://usevia.app/)
```
Comment (waffle87) [inline keyboards/handwired/cans12erv2/readme.md:5]:
```suggestion
```

**Review by waffle87 — APPROVED** [review]

File: keyboards/handwired/cans12erv2/keyboard.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        "pid": "0xABC0",
+        "vid": "0x4342"
+    },
+    "layouts": {
+        "LAYOUT": {
+            "layout": [
+                {"matrix": [0, 0], "x": 0, "y": 0},
+                {"matrix": [0, 1], "x": 1, "y": 0},
+                {"matrix": [0, 2], "x": 2, "y": 0},
+                {"matrix": [0, 3], "x": 3, "y": 0},
```
Comment (dunk2k) [inline keyboards/handwired/cans12erv2/keyboard.json:36]: denotes encoder location
```suggestion
                {"matrix": [0, 3], "x": 3, "y": 0, "encoder": 0},
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/handwired/cans12erv2/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,18 @@
+// Copyright 2024 Can Baytok (https://github.com/canbaytok)
+// SPDX-License-Identifier: GPL-2.0-or-later
+#include QMK_KEYBOARD_H
+#include "keymap_german.h"
```
Comment (drashna) [inline keyboards/handwired/cans12erv2/keymaps/default/keymap.c:4]: This isn't used here, so can/should be removed. 
```suggestion
```

File: keyboards/handwired/cans12erv2/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+        KC_E, KC_F, KC_G, KC_H,
+        KC_I, KC_J, KC_K, KC_L
+    )
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] =   { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) }
+};
+#endif
```
Comment (drashna) [inline keyboards/handwired/cans12erv2/keymaps/default/keymap.c:18]:
```suggestion
#endif

```

File: keyboards/handwired/cans12erv2/keymaps/default/rules.mk [hunk 8]
```diff
@@ -0,0 +1 @@
+ENCODER_MAP_ENABLE = yes
```
Comment (drashna) [inline keyboards/handwired/cans12erv2/keymaps/default/rules.mk:1]:
```suggestion
ENCODER_MAP_ENABLE = yes

```

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request merged**